apps/hash-graph/lib/graph/src/api/rest.rs

```diff
@@ -144,6 +144,20 @@ pub fn rest_api_router<P: StorePool + Send + 'static>(
         )
 }
 
+pub fn openapi_only_router() -> Router {
```

Comment:
Do you think it's worth reusing this snippet?

Reply:
I'm not sure what you mean

Reply:
The function above contains
```rust
    // OpenAPI documentation is also generated by merging resources
    let open_api_doc = OpenApiDocumentation::openapi();
    // super-router can then be used as any other router.
    // Make sure extensions are added at the end so they are made available to merged routers.
    // The `/api-doc` endpoints are nested as we don't want any layers or handlers for the api-doc
    merged_routes
        .layer(Extension(dependencies.store))
        .layer(Extension(dependencies.domain_regex))
        .layer(axum::middleware::from_fn(log_request_and_response))
        .layer(span_trace_layer())
        .nest(
            "/api-doc",
            Router::new()
                .route(
                    "/openapi.json",
                    get({
                        let doc = open_api_doc;
                        move || async { Json(doc) }
                    }),
                )
                .route("/models/*path", get(serve_static_schema)),
         )
```

The new function duplicates the OpenAPI spec stuff. I tested it locally, and it's possible to re-use the new function here as well:

```rust
...
    // super-router can then be used as any other router.
    // Make sure extensions are added at the end so they are made available to merged routers.
    // The `/api-doc` endpoints are nested as we don't want any layers or handlers for the api-doc
    merged_routes
        .layer(Extension(dependencies.store))
        .layer(Extension(dependencies.domain_regex))
        .layer(axum::middleware::from_fn(log_request_and_response))
        .layer(span_trace_layer())
        .merge(openapi_router())

pub fn openapi_router() -> Router {
    // OpenAPI documentation is also generated by merging resources
    let open_api_doc = OpenApiDocumentation::openapi();
...
```

Reply:
Gotcha, I thought it the similar code below didn't take a `Router` but clearly I didn't read well enough, I think it is worth re-using, will do quickly

Reply:
[Reuse openapi_only_router](https://github.com/hashintel/hash/pull/2358/commits/b0f04c1c947054dde6e7c563204545692109278d)